### DISCUSSION TITLE
Importing types

### DISCUSSION THREAD

**Discussion by Fanreza** [epicmaxco/vuestic-ui#3882] [Q&A]
Im wrapping vuestic component with my custom component and use ``v-bind`` to copy all props down, but my custom component doesnt know anything about vuestic props, so is there any exported component props type that i can use?

**Comment by m0ksem** [comment]
Here is an issue for it. We're still thinking about it.

Here is a hint: We do extend components like VaInput in VaDateInput of course, and we use a special utility for it:
https://github.com/epicmaxco/vuestic-ui/blob/develop/packages/ui/src/utils/component-options/extract-component-options.ts

You can take `ExtractComponentProps` type from here https://github.com/epicmaxco/vuestic-ui/blob/develop/packages/ui/src/utils/component-options/types.ts#L35

All you need is:
```ts
import { VaInput } from 'vuestic-ui'

defineProps({
    ...(VaInput.props as ExtractPropTypes<typeof VaInput>)
})
```

It doesn't work with script setup though, but here is my fix, but it is still in progress: https://github.com/epicmaxco/vuestic-ui/pull/3784/files#diff-fa5580e769d658f7b3879158e183c97376772a7240c11307f3bd6e538d906418

Overall, if you have suggestions how we can improve this flow, I would definitely like to implement it.
